<a href="https://colab.research.google.com/github/DinWaikabu/Pandas/blob/master/mengatasi_missing_value.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd

In [0]:
df = pd.read_csv('http://bit.ly/uforeports')

In [3]:
df.head()

City Colors Reported Shape Reported State             Time
0                Ithaca             NaN       TRIANGLE    NY   6/1/1930 22:00
1           Willingboro             NaN          OTHER    NJ  6/30/1930 20:00
2               Holyoke             NaN           OVAL    CO  2/15/1931 14:00
3               Abilene             NaN           DISK    KS   6/1/1931 13:00
4  New York Worlds Fair             NaN          LIGHT    NY  4/18/1933 19:00

In [40]:
df.isnull().head()

City  Colors Reported  Shape Reported  State   Time
0  False             True           False  False  False
1  False             True           False  False  False
2  False             True           False  False  False
3  False             True           False  False  False
4  False             True           False  False  False

In [41]:
df.isnull().sum()

City                  25
Colors Reported    15359
Shape Reported      2644
State                  0
Time                   0
dtype: int64

In [42]:
df[df.isnull()].head()

City Colors Reported Shape Reported State Time
0  NaN             NaN            NaN   NaN  NaN
1  NaN             NaN            NaN   NaN  NaN
2  NaN             NaN            NaN   NaN  NaN
3  NaN             NaN            NaN   NaN  NaN
4  NaN             NaN            NaN   NaN  NaN

###Menghapus Missing Value

In [0]:
df1 = df.copy()

In [16]:
df1.shape

(18241, 5)

sebelum dihapus missing value kita memiliki data sebanyak 18241 baris. kemudian baris yang memiliki missing value dihapus dengan asumsi kita memiliki data yang besar dan data yang telah kita hapus dapat kita jumpai di sisa data lainnya

In [0]:
df1.dropna(inplace=True)

In [23]:
df1.isnull().sum()

City               0
Colors Reported    0
Shape Reported     0
State              0
Time               0
dtype: int64

stelah dihapus, maka tidak ada lagi missing value di dalam dataframe yang kita miliki. 

In [39]:
df1.head()

City Colors Reported Shape Reported State             Time
12      Belton             RED         SPHERE    SC  6/30/1939 20:00
19  Bering Sea             RED          OTHER    AK  4/30/1943 23:00
36  Portsmouth             RED      FORMATION    VA   7/10/1945 1:30
44   Blairsden           GREEN         SPHERE    CA  6/30/1946 19:00
82    San Jose            BLUE        CHEVRON    CA  7/15/1947 21:00